devicetypes/smartenit/smartenit-metering-dual-load-controller.src/smartenit-metering-dual-load-controller.groovy

```diff
@@ -0,0 +1,175 @@
+ import physicalgraph.zigbee.zcl.DataType
+ 
+ @Field final CurrentLevel = 0x0000
+ @Field final MoveToLevelWOnOff = 0x0004
+ @Field final MeteringCurrentSummation = 0x0000
+ @Field final MeteringInstantDemand = 0x0400
+ @Field final EnergyDivisor = 100000
+ @Field final CurrentDivisor = 100
+ @Field final Current = 0x00f0
+ @Field final Voltage = 0x00f1
+ @Field final OnOff = 0x0000
+ @Field final SmartenitMfrCode = 0x1075
+
+metadata {
+	definition (name: "Smartenit Metering Dual Load Controller", namespace: "Smartenit", author: "Luis Contreras", mnmn: "SmartThingsCommunity", vid: "472dac67-bbdd-344e-944b-43abafeeb82b") {
+		capability "Actuator"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Power Meter"
+		capability "Energy Meter"
+		capability "Health Check"
+		capability "Voltage Measurement"
+		capability "monthpublic25501.current"
+		capability "monthpublic25501.load1"
+		capability "monthpublic25501.load2"
+		capability "monthpublic25501.levelControl"
+
+		fingerprint model: "ZBMLC30NC", manufacturer: "Smartenit, Inc", deviceJoinName: "Smartenit Dual Switch Meter"
+		fingerprint model: "ZBMLC30NO", manufacturer: "Smartenit, Inc", deviceJoinName: "Smartenit Dual Switch Meter"
+	}
+}
+
+def getFPoint(String FPointHex){
+	return (Float)Long.parseLong(FPointHex, 16)
+}
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "Basic description: ${description}"
+	def event = zigbee.getEvent(description)
+	Map eventDescMap = zigbee.parseDescriptionAsMap(description)
+
+	if (description?.startsWith("on/off")) {
+		def cmds = zigbee.readAttribute(zigbee.ONOFF_CLUSTER, OnOff) + zigbee.readAttribute(zigbee.ONOFF_CLUSTER, OnOff, [destEndpoint: 2])
+		return cmds.collect { new physicalgraph.device.HubAction(it) }
+	}
+
+	if (event && event.name != "switch") {
+		log.debug "event: ${event}, ${event.name}, ${event.value}"
+		if ((eventDescMap?.sourceEndpoint == "01") || (eventDescMap?.endpoint == "01")) {
+			if (event.name == "power") {
+				sendEvent(name: "power", value: (event.value/EnergyDivisor))
+			} else {
+				sendEvent(event) 
+			}
+		}
+	} else {
+		def mapDescription = zigbee.parseDescriptionAsMap(description)
+		log.debug "mapDescription... : ${mapDescription}"
+
+		if (mapDescription) {
+			if (mapDescription.clusterInt == zigbee.SIMPLE_METERING_CLUSTER) {
+				if (mapDescription.attrInt == MeteringCurrentSummation) {
+					return sendEvent(name:"energy", value: getFPoint(mapDescription.value)/EnergyDivisor)
+				} else if (mapDescription.attrInt == MeteringInstantDemand) {
+					return sendEvent(name:"power", value: getFPoint(mapDescription.value/EnergyDivisor))
+				} else if (mapDescription.attrInt == Voltage) {
+					return sendEvent(name:"voltage", value: getFPoint(mapDescription.value) / 100)
+				} else if (mapDescription.attrInt == Current) {
+					return sendEvent(name:"current", value: getFPoint(mapDescription.value) / 100, unit: "A")
+				}
```

Comment:
If you want to return an event, it is more correct to use `createEvent()`. It takes same arguments as `sendEvent()`.
Main difference between them is that `createEvent()` returns an event object which needs to be returned from `parse()` method to be acknowledged by the platform, while `sendEvent()` sends an event immediately and returns no object (as far as I can tell).